src/pylexibank/commands/format_profile.py

```diff
@@ -0,0 +1,151 @@
+"""
+
+"""
+import re
+import collections
+
+import pyclts
+from segments import Tokenizer
+
+from cldfbench.cli_util import add_catalog_spec, get_dataset
+from pylexibank.cli_util import add_dataset_spec
+from pylexibank.profile import IPA
+
+
+def register(parser):
+    add_dataset_spec(parser)
+    add_catalog_spec(parser, 'clts')
+
+    parser.add_argument(
+        "--ipa",
+        help="Name of the IPA column (default: `IPA`).",
+        default="IPA",
+    )
+    parser.add_argument(
+        '--augment',
+        help="augment the profile with frequency counts and examples from the FormTable",
+        default=False,
+        action='store_true',
+    )
+    parser.add_argument(
+        '--trim',
+        help="remove redundant orthography rules",
+        default=False,
+        action='store_true',
+    )
+    parser.add_argument(
+        '--sort',
+        help="sort the profile",
+        default=False,
+        action='store_true',
+    )
+
+
+def run(args):
+    ds = get_dataset(args)
+    clts = args.clts.api
+
+    # Load the profile(s) specified for the dataset
+    profiles = {k or 'default': v for k, v in ds.orthography_profile_dict.items()}
+    forms = collections.defaultdict(list)
+    if ds.cldf_dir.joinpath('forms.csv').exists():
+        for form in ds.cldf_reader()['FormTable']:
+            forms[form.get('Profile')].append(ds.form_for_segmentation(form['Form']))
+    if list(forms.keys()) == [None]:
+        forms['default'] = forms[None]
+
+    for key, profile in profiles.items():
+        args.log.info('Processing {0}'.format(profile.fname))
+        clean_profile(profile, clts, args)
+
+        if args.trim:
+            # Run the trimmer as many times as necessary until nothing more is left to remove
+            total_removed = 0
+            while True:
+                removed = profile.trim(ipa_col=args.ipa)
+                total_removed += removed
+                if removed == 0:
+                    break
+            if total_removed:
+                args.log.info("%i superfluous rules were removed.", total_removed)
+
+        if args.augment and forms[key]:
+            profile.augment(forms[key])
+
+        if args.sort:
+            profile.sort(clts=args.clts.api, ipa_col=args.ipa)
+
+        check_consistency(profile, clts, args)
+
+        profile.fname.write_text(str(profile), encoding='utf8')
+
+
+def unicode2codepointstr(text):
+    """
+    Returns a codepoint representation to an Unicode string.
+    """
+    return " ".join(["U+{0:0{1}X}".format(ord(char), 4) for char in text])
+
+
+def check_consistency(profile, clts, args):
```

Comment:
@tresoldi `check_consistency` and `clean_profile` will be moved to methods of `pylexibank.profile.Profile` as well.

Reply:
:+1: